drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -363,31 +364,88 @@ impl<S: SpiServer> Io<S> {
 
     // Poll ROT_IRQ until asserted (true) or deasserted (false).
     //
-    // We sleep and poll for what should be long enough for the RoT to queue
-    // a response.
-    //
-    // TODO: Use STM32 EXTI as  an interrupt allows for better performance and
-    // power efficiency.
-    //
-    // STM32 EXTI allows for 16 interrupts for GPIOs.
-    // Each of those can represent Pin X from a GPIO bank (A through K)
-    // So, only one bank's Pin 3, for example, can have the #3 interrupt.
-    // For ROT_IRQ, we would configure for the falling edge to trigger
-    // the interrupt. That configuration should be specified in the app.toml
-    // for the board. Work needs to be done to generalize the EXTI facility.
-    // But, hacking in one interrupt as an example should be ok to start things
-    // off.
+    // We do this by asking the `sys` task to notify us when the GPIO pin's
+    // state changes (using EXTI), and waiting for either that or timeout
+    // determined based on `max_sleep`.
     fn wait_rot_irq(&mut self, desired: bool, max_sleep: u32) -> bool {
-        let mut slept = 0;
+        // Determine our edge sensitivity for the interrupt. If we want to wait
+        // for the line to be asserted, we want to wait for a rising edge. If
+        // the line is currently asserted, and we're waiting for it to be
+        // *deasserted*, we want to wait for a falling edge.
+        let sensitivity = match desired {
+            true => sys_api::Edge::Rising,
+            false => sys_api::Edge::Falling,
+        };
+        self.sys
+            .gpio_irq_configure(notifications::ROT_IRQ_MASK, sensitivity);
+
+        // Enable the interrupt.
+        self.sys
+            .gpio_irq_control(notifications::ROT_IRQ_MASK, IrqControl::Enable)
+            // Just unwrap this, because the `sys` task should never panic.
+            .unwrap_lite();
+
+        // Determine the deadline after which we'll give up, and start the clock.
+        let deadline = sys_get_timer()
+            .now
+            .checked_add(max_sleep as u64)
```

Comment:
All of the callers are using constants for max_sleep IIRC which means we're probably safe not testing for very large max_sleep.
Should this be .saturating_add in case we start testing with clock close to roll-over?
The down-side would be a timeout that is too short for that close-to-rollover clock, but this task would retry.